theano/tensor/slinalg.py

```diff
@@ -413,7 +413,12 @@ def perform(self, node, inputs, outputs):
         U = scipy.linalg.inv(V).T
 
         exp_w = numpy.exp(w)
-        X = numpy.subtract.outer(exp_w, exp_w) / numpy.subtract.outer(w, w)
+        exp_diff = numpy.subtract.outer(exp_w, exp_w)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", RuntimeWarning)
+            X = exp_diff/numpy.subtract.outer(w, w)
+        X[exp_diff == 0] = 0
```

Comment:
This line change the results. Can you add a test that cover this case? It will make sure it work well and don't get broken in the futur.

thanks